src/wallet/use-cases/register-wallet/register-wallet.spec.ts

```diff
@@ -0,0 +1,120 @@
+  let walletRepository: IWalletRepository;
+  let listUserByIdUseCase: ListUserByIdUseCase;
+  let findBankByIdUseCase: FindBankByIdUseCase;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        RegisterWalletUseCase,
+        {
+          provide: IWalletRepository,
+          useValue: {
+            create: jest.fn(),
+          },
+        },
+        {
+          provide: ListUserByIdUseCase,
+          useValue: {
+            execute: jest.fn(),
+          },
+        },
+        {
+          provide: FindBankByIdUseCase,
+          useValue: {
+            execute: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    registerWalletUseCase = module.get<RegisterWalletUseCase>(
+      RegisterWalletUseCase,
+    );
+    walletRepository = module.get<IWalletRepository>(IWalletRepository);
+    listUserByIdUseCase = module.get<ListUserByIdUseCase>(ListUserByIdUseCase);
+    findBankByIdUseCase = module.get<FindBankByIdUseCase>(FindBankByIdUseCase);
+  });
+
+  it('should be able to register a new wallet', async () => {
+    listUserByIdUseCase.execute = jest.fn().mockResolvedValueOnce({});
+    findBankByIdUseCase.execute = jest.fn().mockResolvedValueOnce({});
+    walletRepository.create = jest.fn().mockResolvedValue({} as Wallet);
+
+    const saveWalletDTO: SaveWalletDTO = {
+      user_id: 'user_id',
+      bank_id: 'bank_id',
+      account_type: AccountType.CHECKING_ACCOUNT,
+      description: 'Descrição da carteira',
+    };
+
+    const result = await registerWalletUseCase.createWallet(saveWalletDTO);
+
+    expect(result).toBeDefined();
+    expect(walletRepository.create).toHaveBeenCalledWith(saveWalletDTO);
+  });
+
+  it('should not be able to return a wallet when bank id do not exist', async () => {
+    listUserByIdUseCase.execute = jest.fn().mockResolvedValueOnce({});
+    findBankByIdUseCase.execute = jest.fn().mockResolvedValueOnce(null);
+
+    const saveWalletDTO: SaveWalletDTO = {
+      user_id: 'user_id',
+      bank_id: 'bank_id',
+      account_type: AccountType.CHECKING_ACCOUNT,
+      description: 'Wallet description',
+    };
+
+    await expect(
+      registerWalletUseCase.createWallet(saveWalletDTO),
+    ).rejects.toThrowError(NotFoundException);
+  });
+
+  it('should not be able to return a wallet when user id do not exist', async () => {
+    listUserByIdUseCase.execute = jest.fn().mockResolvedValueOnce(undefined);
+    findBankByIdUseCase.execute = jest.fn().mockResolvedValueOnce({});
+
+    const saveWalletDTO: SaveWalletDTO = {
+      user_id: 'user_id',
+      bank_id: 'bank_id',
+      account_type: AccountType.CHECKING_ACCOUNT,
+      description: 'Wallet description',
+    };
+
+    await expect(
+      registerWalletUseCase.createWallet(saveWalletDTO),
+    ).rejects.toThrowError(NotFoundException);
+  });
+
+  it('should not be able to return a wallet if wallet register fails', async () => {
```

Comment:
Coloca um "the" só para deixar claro que estamos nos referindo a registro da carteira.

"should not be able to return a wallet if the wallet register fails."

Reply:
Arrumei!